src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletControllerTest.java

```diff
@@ -1,7 +1,88 @@
 package pl.byczazagroda.trackexpensesappbackend.controller;
 
-import static org.junit.jupiter.api.Assertions.*;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.ResultActions;
+import pl.byczazagroda.trackexpensesappbackend.BaseControllerTest;
+import pl.byczazagroda.trackexpensesappbackend.dto.CreateWalletDTO;
+import pl.byczazagroda.trackexpensesappbackend.dto.WalletDTO;
+import pl.byczazagroda.trackexpensesappbackend.service.WalletServiceImpl;
+
+import java.time.Instant;
+import java.util.Objects;
+
+import static org.mockito.BDDMockito.given;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+class WalletControllerTest extends BaseControllerTest {
+
+    @Mock
+    private WalletServiceImpl walletService;
+
+
+    @Test
+    void shouldThrowAnExceptionWhenNameIsEmpty() throws Exception {
```

Comment:
As it seems that all controller tests are similiar then maybe we can use parameterized tests cases here?

My proposition is:

```
    @DisplayName("Should throw an exception when wallet name is null, empty, contains more than 20 character or contains illegal letters")
    @ParameterizedTest
    @NullAndEmptySource
    @ValueSource(strings = {"This wallet name is too long, it contains over 20 characters", "#$@"})
    void shouldThrowAnExceptionWhenWalletNameIsInvalid(String walletName) throws Exception {
        // given
        CreateWalletDTO createWalletDTO = new CreateWalletDTO(walletName);
        given(walletService.createWallet(createWalletDTO))
                .willReturn(new WalletDTO(1L, walletName, Instant.now()));

        // when
        ResultActions result = mockMvc.perform(post("/api/wallet")
                .contentType(MediaType.APPLICATION_JSON)
                .content(Objects.requireNonNull(objectMapper.writeValueAsString(createWalletDTO))));
        // then
        result.andExpect(status().isBadRequest());
    }
```
Sonarcloud also pointed it out.